src/loaders/MatrixZarrLoader.js

```diff
@@ -0,0 +1,56 @@
+import { openArray } from 'zarr';
+import AbstractLoader from './AbstractLoader';
+
+function basename(path) {
+  return path.split('/').pop();
+}
+
+function dirname(path) {
+  return path.substr(0, path.lastIndexOf('/') + 1);
+}
+
+export default class MatrixZarrLoader extends AbstractLoader {
+  constructor(params) {
+    super(params);
+
+    const { url } = this;
+    this.zUrl = url.endsWith('/') ? url : `${url}/`;
+    this.zAttrsUrl = `${this.zUrl}.zattrs`;
```

Comment:
I'm curious what the hierarchy is for this zarr. I mostly am curious because by dealing with the URL (and not utilizing a store), it prevents from using a non-HTTP store for zarr. Not really of concern at the moment, but something I've run into with jupyter integration and Viv. Happy to help troubleshoot.

Reply:
I referenced the attribute-fetching code from Viv when writing this. https://github.com/hubmapconsortium/vitessce-image-viewer/blob/921bb5e594075e75190e215a7dfba9bae5e87732/src/loaders/omeZarrReader.js#L36 Is there a better way to access `.attrs`?

Reply:
Yeah sorry, this is a hack. `.attrs` _should_ be a property on the underlying zarr object (group or array) but it's a bug in Zarr.js right now because the `ATTRS_META_KEY` isn't correct for the new zarr spec:

https://github.com/gzuidhof/zarr.js/blob/master/src/names.ts#L3

Is the zarr a hierarchy (group with nested arrays) or just an array?

However, in the meantime, what I'd suggest is creating an `HTTPStore` from the URL (which is done internally if you provide a string as a store in `openArray`). This means if in the future there is a different store (i.e. some class that wraps the imjoy-rpc)`, the code is reusable.

Metadata are always JSON blobs (ascii encoded), so you can parse the metadata like this:


```javascript
async function getJson(store, key) {
  const bytes = new Uint8Array(await store.getItem(key));
  const decoder = new TextDecoder('utf-8');
  const json = JSON.parse(decoder.decode(bytes));
  return json;
}

import { HTTPStore, openArray } from 'zarr';

let store = "http://localhost:8000/mydata.zarr"
// or store = someSpecialStore();
if (typeof store === "string") {
  store = new HTTPStore(url);
}
const attrs = await getJson(store, '.zattrs');
const z = await openArray({ store, path: 'path/to/array/' });
```

Reply:
I have updated to use this method, thanks!